deltachat-ffi/src/lib.rs

```diff
@@ -2353,6 +2353,27 @@ pub unsafe extern "C" fn dc_chat_is_sending_locations(chat: *mut dc_chat_t) -> l
     ffi_chat.chat.is_sending_locations() as libc::c_int
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn dc_chat_get_info_json(
+    context: *mut dc_context_t,
+    chat_id: u32,
+) -> *mut libc::c_char {
+    if context.is_null() {
+        eprintln!("ignoring careless call to dc_get_oauth2_url()");
+        return ptr::null_mut(); // NULL explicitly defined as "unknown"
+    }
+    let ffi_context = &*context;
+    ffi_context
+        .with_inner(|ctx| match chat::get_info_json(ctx, chat_id) {
+            Ok(s) => s.strdup(),
+            Err(err) => {
+                error!(ctx, "get_info_json({}) returned: {}", chat_id, err);
+                "".strdup()
+            }
+        })
+        .unwrap_or_else(|_| ptr::null_mut())
```

Comment:
docs says NULL is never returned.  Yet here it is.
Looking at the implementation of .with_inner() this would actually be returned if the context is not open, so this probably matters and should also be `"".strdup()`?

Reply:
Not sure what return type makes sense if we expect a json... Will evaluate this as soon as i'm working on the desktop side.

Reply:
in general, i tend to return the empty string instead of NULL if there is just an error and the error cases are rare - these things tend to be overseen/ignored in c when usually not expected.

only when NULL is a very common case, i returned it explicitly.

not sure if that makes sense here, though :)

Reply:
@r10s yup makes sense here, will change it to an empty string :)